### ISSUE TITLE
salt client api causing ioloop.py issue

### ISSUE THREAD

**Issue by harishkapri** [saltstack/salt#56577]
### Description of Issue
**BEFORE:**
I was using salt-master with **Salt Version:Salt: 2015.8.7** and **salt-minion 2015.8.7 (Beryllium)** and no issues encountered and was able to use salt client api without any issues.
**AFTER:**
I set up new salt-master with **Salt Version:Salt: 2017.7.8** and **salt-minion 2017.7.8 (Nitrogen)**
I am unable to use the salt client api which is required by our custom scripts which was working fine in the previous version of salt and not working with the newer version of our salt setup. Any help is appreciated as we are unable to make salt client api calls due to this error.

```
ERROR:tornado.application:Exception in callback <functools.partial object at 0x7feee5624208>
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/tornado/ioloop.py", line 592, in _run_callback
    ret = callback()
  File "/usr/lib/python2.7/dist-packages/tornado/stack_context.py", line 275, in null_wrapper
    return fn(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/tornado/ioloop.py", line 598, in <lambda>
    self.add_future(ret, lambda f: f.result())
  File "/usr/lib/python2.7/dist-packages/tornado/concurrent.py", line 215, in result
    raise_exc_info(self._exc_info)
  File "/usr/lib/python2.7/dist-packages/tornado/gen.py", line 230, in wrapper
    yielded = next(result)
  File "/usr/lib/python2.7/dist-packages/salt/crypt.py", line 510, in _authenticate
    io_loop=self.io_loop)
  File "/usr/lib/python2.7/dist-packages/salt/transport/client.py", line 108, in factory
    return salt.transport.zeromq.AsyncZeroMQReqChannel(opts, **kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/transport/zeromq.py", line 130, in __new__
    obj.__singleton_init__(opts, **kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/transport/zeromq.py", line 193, in __singleton_init__
    kwargs={'io_loop': self._io_loop})
  File "/usr/lib/python2.7/dist-packages/salt/transport/zeromq.py", line 907, in __init__
    super(AsyncReqMessageClientPool, self).__init__(AsyncReqMessageClient, opts, args=args, kwargs=kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/transport/__init__.py", line 69, in __init__
    self.message_clients = [tgt(*args, **kwargs) for _ in range(sock_pool_size)]
  File "/usr/lib/python2.7/dist-packages/salt/transport/zeromq.py", line 954, in __init__
    self._init_socket()
  File "/usr/lib/python2.7/dist-packages/salt/transport/zeromq.py", line 1007, in _init_socket
    self.stream = zmq.eventloop.zmqstream.ZMQStream(self.socket, io_loop=self.io_loop)
  File "/usr/lib/python2.7/dist-packages/zmq/eventloop/zmqstream.py", line 114, in __init__
    self._init_io_state()
  File "/usr/lib/python2.7/dist-packages/zmq/eventloop/zmqstream.py", line 535, in _init_io_state
    self.io_loop.add_handler(self.socket, self._handle_events, self._state)
  File "/usr/lib/python2.7/dist-packages/tornado/ioloop.py", line 704, in add_handler
    self._impl.register(fd, events | self.ERROR)
TypeError: argument must be an int, or have a fileno() method
```

### Setup
**pip freeze output**
```
amibuilder==1.1.7
apache-libcloud==1.5.0
awscli==1.18.17
backports-abc==0.5
beanstalkc==0.4.0
beautifulsoup4==4.4.1
boto==2.49.0
boto3==1.10.33
botocore==1.13.33
certifi==2019.11.28
cffi==1.14.0
chardet==3.0.4
CherryPy==3.5.0
Click==7.0
colorama==0.4.3
configobj==5.0.6
coverage==5.0.3
cryptography==2.8
decorator==4.4.2
docutils==0.15.2
easgpolicy==0.0.1
ecsdeploy==0.0.4
edeploy==1.2.14
elib==1.5.4
elist==1.0.0
enum34==1.1.2
envoy==0.0.3
Flask==1.1.1
futures==3.3.0
gitdb==0.6.4
GitPython==1.0.1
html5lib==0.999
idna==2.9
ipaddress==1.0.16
itsdangerous==1.1.0
Jinja2==2.11.1
jmespath==0.9.5
linecache2==1.0.0
lockfile==0.12.2
lxml==3.5.0
M2Crypto==0.22.6rc4
Mako==1.0.3
MarkupSafe==1.1.1
mock==4.0.1
msgpack==0.6.2
ndg-httpsclient==0.4.0
netaddr==0.7.18
nodelib==1.3.2
nose==1.3.7
psutil==3.4.2
py==1.8.1
py-salt==1.0
py-zabbix==1.1.5
pyaml==20.3.1
pyasn1==0.4.8
pycparser==2.20
pycrypto==2.6.1
pycurl==7.43.0
PyMySQL==0.7.2
pyOpenSSL==19.1.0
python-apt==1.1.0b1+ubuntu0.16.4.8
python-dateutil==2.8.1
python-gnupg==0.3.8
python-systemd==231
PyYAML==5.3
pyzmq==15.2.0
repoze.lru==0.6
requests==2.23.0
retry==0.9.2
Routes==2.2
rsa==3.4.2
rundeckutils==1.0.2
s3transfer==0.2.1
salt==2015.8.10
simplejson==3.8.1
singledispatch==3.4.0.3
six==1.14.0
smmap==0.9.0
statsd==3.3.0
tabulate==0.8.6
tornado==4.2.1
traceback2==1.4.0
unittest2==1.1.0
UNKNOWN==4.0.1
urllib3==1.25.8
virtualenv==15.0.1
WebOb==1.5.1
Werkzeug==1.0.0
wheel==0.29.0

```
### Versions Report
```
root@salt-upgraded:~$ salt --versions-report
Salt Version:
           Salt: 2017.7.8

Dependency Versions:
           cffi: 1.14.0
       cherrypy: 3.5.0
       dateutil: 2.8.1
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.11.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.6.2
   mysql-python: Not Installed
      pycparser: 2.20
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Oct  8 2019, 14:14:10)
   python-gnupg: 0.3.8
         PyYAML: 5.3
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-1102-aws
         system: Linux
        version: Ubuntu 16.04 xenial

```
```
salt-minion --version
salt-minion 2017.7.8 (Nitrogen)
```

**Comment by sagetherage** [comment]
similar issue: https://github.com/saltstack/salt/issues/50327

**Comment by harishkapri** [comment]
@sagetherage sorry to bother, any updates on when the fix will be released.

**Comment by ManhVDb** [comment]
i want to update file in service by api , help me keyword

**Comment by dwoz** [comment]
@ManhVDb I think this would be resolved by upgrading tornado to 4.5.3. But also, if you are upgrading salt you should move to a currently supported version (preferably 3002.2).

**Comment by sagetherage** [comment]
closing as won't fix in these unsupported Salt Versions
